swiftide-integrations/src/pgvector/pgv_table_types.rs

```diff
@@ -0,0 +1,417 @@
+        let columns: Vec<String> = self
+            .fields
+            .iter()
+            .map(|field| match field {
+                FieldConfig::ID => "id UUID NOT NULL".to_string(),
+                FieldConfig::Chunk => format!("{} TEXT NOT NULL", field.field_name()),
+                FieldConfig::Metadata(_) => format!("{} JSONB", field.field_name()),
+                FieldConfig::Vector(_) => format!("{} VECTOR({})", field.field_name(), vector_size),
+            })
+            .chain(std::iter::once("record_id SERIAL PRIMARY KEY".to_string()))
+            .collect();
+
+        let sql = format!(
+            "CREATE TABLE IF NOT EXISTS {} (\n  {}\n)",
+            self.table_name,
+            columns.join(",\n  ")
+        );
+
+        Ok(sql)
+    }
+
+    /// Generates the SQL statement to create an HNSW index on the vector column.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if:
+    /// - No vector field is found in the table configuration.
+    /// - The table name or field name is invalid.
+    pub fn create_index_sql(&self) -> Result<String> {
+        let index_name = format!("{}_embedding_idx", self.table_name);
+        let vector_field = self
+            .fields
+            .iter()
+            .find(|f| matches!(f, FieldConfig::Vector(_)))
+            .ok_or_else(|| anyhow::anyhow!("No vector field found in configuration"))?
+            .field_name();
+
+        // Validate table_name and field_name (e.g., check against allowed patterns)
+        if !Self::is_valid_identifier(&self.table_name)
+            || !Self::is_valid_identifier(&index_name)
+            || !Self::is_valid_identifier(vector_field)
+        {
+            return Err(anyhow::anyhow!("Invalid table or field name"));
+        }
+
+        Ok(format!(
+            "CREATE INDEX IF NOT EXISTS {} ON {} USING hnsw ({} vector_cosine_ops)",
+            index_name, &self.table_name, vector_field
+        ))
+    }
+
+    /// Stores a list of nodes in the database using an upsert operation.
+    ///
+    /// # Arguments
+    ///
+    /// * `nodes` - A slice of `Node` objects to be stored.
+    ///
+    /// # Returns
+    ///
+    /// * `Result<()>` - Ok if all nodes are successfully stored, Err otherwise.
+    ///
+    /// # Errors
+    ///
+    /// This function will return an error if:
+    /// - The database connection pool is not established.
+    /// - Any of the SQL queries fail to execute.
+    /// - Committing the transaction fails.
+    pub async fn store_nodes(&self, nodes: &[Node]) -> Result<()> {
+        let pool = self.connection_pool.get_pool()?;
+
+        let mut tx = pool.begin().await?;
+        let sql = self.generate_bulk_insert_sql(nodes.len());
+
+        let mut query = sqlx::query(&sql);
+
+        for node in nodes {
+            query = self.bind_node_to_bulk_query(query, node)?;
+        }
+
+        query.execute(&mut *tx).await.map_err(|e| {
+            tracing::error!("Failed to store nodes: {:?}", e);
```

Comment:
It's good practice not to log errors in libraries if the error is bubbled up, as the caller is expected to handle it. Could you remove it?

Reply:
Sounds good! I've cleaned up the extra logs and traces.